engine/src/main/java/org/terasology/i18n/TranslationSystemImpl.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.i18n;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.config.Config;
+import org.terasology.config.SystemConfig;
+import org.terasology.context.Context;
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.Uri;
+import org.terasology.i18n.assets.Translation;
+
+/**
+ * A translation system that uses {@link Translation} data assets to
+ * perform the lookup.
+ */
+public class TranslationSystemImpl implements TranslationSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(TranslationSystemImpl.class);
+
+    private final List<Consumer<TranslationProject>> changeListeners = new CopyOnWriteArrayList<>();
+    private final Map<Uri, TranslationProject> projects = new HashMap<>();
+
+    private final SystemConfig config;
+
+    /**
+     * @param context the context to use
+     */
+    public TranslationSystemImpl(Context context) {
+
+        config = context.get(Config.class).getSystem();
+        AssetManager assetManager = context.get(AssetManager.class);
+
+        Set<ResourceUrn> urns = assetManager.getAvailableAssets(Translation.class);
```

Comment:
I wonder if this needs to be done upfront, or could be done on demand. All the information needed to discover the relevant asset is available from a translation uri + locale, so rather than loading them all and indexing, could just search for them as needed?